tests/Unit/DataStructures/Tensor/EagerMath/Test_CrossProduct.cpp

```diff
@@ -0,0 +1,209 @@
+      [&two, &minus_three, &four, &minus_five, &minus_twelve,
+       &minus_thirteen]() {
+        tnsr::II<DataType, 3, Frame::Grid> tensor{};
+        get<0, 0>(tensor) = two;
+        get<0, 1>(tensor) = minus_three;
+        get<0, 2>(tensor) = four;
+        get<1, 1>(tensor) = minus_five;
+        get<1, 2>(tensor) = minus_twelve;
+        get<2, 2>(tensor) = minus_thirteen;
+        return tensor;
+      }();
+
+  const Scalar<DataType> det_metric = [&one_over_three_hundred_twenty_seven]() {
+    Scalar<DataType> scalar{};
+    get(scalar) = one_over_three_hundred_twenty_seven;
+    return scalar;
+  }();
+
+  // The known case for curved-space happens to involve the same vector_a
+  // but different vector_b as for the Euclidean cross product test
+  const tnsr::I<DataType, 3, Frame::Grid> curved_vector_b{
+      {{minus_five, four, two}}};
+  const tnsr::i<DataType, 3, Frame::Grid> curved_covector_b =
+      [&fifty_three, &ninety_seven, &minus_ninety]() {
+        tnsr::i<DataType, 3, Frame::Grid> tensor{};
+        get<0>(tensor) = fifty_three / 109.0;
+        get<1>(tensor) = ninety_seven / 109.0;
+        get<2>(tensor) = minus_ninety / 109.0;
+        return tensor;
+      }();
+
+  const tnsr::I<DataType, 3, Frame::Grid> curved_vector_expected =
+      [&two_hundred_fifty, &minus_five_hundred_thirty,
+       &minus_four_hundred_twenty_four, &det_metric]() {
+        tnsr::I<DataType, 3, Frame::Grid> tensor{};
+        get<0>(tensor) = two_hundred_fifty * sqrt(get(det_metric));
+        get<1>(tensor) = minus_five_hundred_thirty * sqrt(get(det_metric));
+        get<2>(tensor) = minus_four_hundred_twenty_four * sqrt(get(det_metric));
+        return tensor;
+      }();
+  const tnsr::i<DataType, 3, Frame::Grid> curved_covector_expected =
+      [&eight, &minus_fourteen, &forty_eight, &det_metric]() {
+        tnsr::i<DataType, 3, Frame::Grid> tensor{};
+        get<0>(tensor) = eight * sqrt(get(det_metric));
+        get<1>(tensor) = minus_fourteen * sqrt(get(det_metric));
+        get<2>(tensor) = forty_eight * sqrt(get(det_metric));
+        return tensor;
+      }();
+
+  CHECK_ITERABLE_APPROX(
+      cross_product(vector_a, curved_vector_b, inverse_metric, det_metric),
+      curved_vector_expected);
+  CHECK_ITERABLE_APPROX(
+      cross_product(vector_a, curved_covector_b, inverse_metric, det_metric),
+      curved_covector_expected);
+
+  // Test Euclidean cross product for a known simple case
+  const tnsr::I<DataType, 3, Frame::Grid> vector_x_hat{{{one, zero, zero}}};
+  const tnsr::I<DataType, 3, Frame::Grid> vector_y_hat{{{zero, one, zero}}};
+  const tnsr::I<DataType, 3, Frame::Grid> covector_y_hat{{{zero, one, zero}}};
+  const tnsr::I<DataType, 3, Frame::Grid> vector_z_hat{{{zero, zero, one}}};
+  const tnsr::I<DataType, 3, Frame::Grid> covector_z_hat{{{zero, zero, one}}};
+  CHECK_ITERABLE_APPROX(cross_product(vector_x_hat, vector_y_hat),
+                        vector_z_hat);
```

Comment:
I suggest adding one "opposite parity" case, e.g., `y cross x -> -z` or `z cross y -> -x`. Also for covectors below.